src/main/java/net/dv8tion/jda/api/entities/RoleConnectionMetadata.java

```diff
@@ -0,0 +1,398 @@
+     * @param  description
+     *         The translated description to put
+     *
+     * @throws IllegalArgumentException
+     *         <ul>
+     *             <li>If the locale is null</li>
+     *             <li>If the description is null</li>
+     *             <li>If the locale is {@link DiscordLocale#UNKNOWN}</li>
+     *             <li>If the provided description is empty or more than {@value MAX_DESCRIPTION_LENGTH} characters long</li>
+     *         </ul>
+     *
+     * @return This updated record instance
+     */
+    @Nonnull
+    public RoleConnectionMetadata setDescriptionLocalization(@Nonnull DiscordLocale locale, @Nonnull String description)
+    {
+        this.descriptionLocalization.setTranslation(locale, description);
+        return this;
+    }
+
+    /**
+     * Sets multiple {@link DiscordLocale language-specific} localizations of this record's description.
+     *
+     * <p>This change will not take effect in Discord until you update the role connection metadata using {@link JDA#updateRoleConnectionMetadata(Collection)}.
+     *
+     * @param  map
+     *         The map from which to transfer the translated descriptions
+     *
+     * @throws IllegalArgumentException
+     *         <ul>
+     *             <li>If the map is null</li>
+     *             <li>If the map contains an {@link DiscordLocale#UNKNOWN} key</li>
+     *             <li>If the map contains a description which is empty or more than {@value MAX_DESCRIPTION_LENGTH} characters long</li>
+     *         </ul>
+     *
+     * @return This updated record instance
+     */
+    @Nonnull
+    public RoleConnectionMetadata setDescriptionLocalizations(@Nonnull Map<DiscordLocale, String> map)
+    {
+        this.descriptionLocalization.setTranslations(map);
+        return this;
+    }
+
+    @Override
+    public String toString()
+    {
+        return new EntityString(this)
+            .setType(type)
+            .addMetadata("key", key)
+            .addMetadata("name", name)
+            .toString();
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (!(obj instanceof RoleConnectionMetadata))
+            return false;
+        if (this == obj)
+            return true;
+        RoleConnectionMetadata o = (RoleConnectionMetadata) obj;
+        return this.type == o.type
+            && this.key.equals(o.key)
+            && this.name.equals(o.name)
+            && this.description.equals(o.description);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(type, key, name, description);
+    }
+
+    @Nonnull
+    @Override
+    public DataObject toData()
+    {
+        return DataObject.empty()
+            .put("type", type.value)
+            .put("name", name)
+            .put("key", key)
+            .put("description", description)
+            .put("name_localizations", nameLocalization)
+            .put("description_localizations", descriptionLocalization);
```

Comment:
These will potentially send empty maps. Not sure if that's a problem for updates

Reply:
We assume updates are always replacing